dbt/adapters/duckdb/plugins/delta.py

```diff
@@ -43,6 +44,110 @@ def load(self, source_config: SourceConfig):
+        table_path = target_config.location.path
+        storage_options = target_config.config.get("storage_options", {})
+
+        if mode == "overwrite_partition":
+            partition_key = target_config.config.get("partition_key", None)
+            if not partition_key:
+                raise Exception(
+                    "'partition_key' has to be defined when mode 'overwrite_partition'!"
+                )
+
+            if isinstance(partition_key, str):
+                partition_key = [partition_key]
+
+            partition_dict = []
+            for each_key in partition_key:
+                unique_key_array = pc.unique(df[each_key])
+
+                if len(unique_key_array) == 1:
+                    partition_dict.append((each_key, str(unique_key_array[0])))
+                else:
+                    raise Exception(
+                        f"'{each_key}' column has not one unique value, values are: {str(unique_key_array)}"
+                    )
+            create_insert_partition(table_path, df, partition_dict, storage_options)
+        elif mode == "merge":
+            # very slow -> https://github.com/delta-io/delta-rs/issues/1846
+            unique_key = target_config.config.get("unique_key", None)
+            if not unique_key:
+                raise Exception("'unique_key' has to be defined when mode 'merge'!")
+            if isinstance(unique_key, str):
+                unique_key = [unique_key]
+
+            predicate_stm = " and ".join(
+                [
+                    f'source."{each_unique_key}" = target."{each_unique_key}"'
+                    for each_unique_key in unique_key
+                ]
+            )
+
+            try:
+                target_dt = DeltaTable(table_path, storage_options=storage_options)
+            except Exception:
+                # TODO handle this better
+                write_deltalake(
+                    table_or_uri=table_path, data=df, storage_options=storage_options
+                )
+
+            target_dt = DeltaTable(table_path, storage_options=storage_options)
+            # TODO there is a problem if the column name is uppercase
+            target_dt.merge(
+                source=df,
+                predicate=predicate_stm,
+                source_alias="source",
+                target_alias="target",
+            ).when_not_matched_insert_all().execute()
+        else:
+            write_deltalake(
+                table_or_uri=table_path,
+                data=df,
+                mode=mode,
+                storage_options=storage_options,
+            )
+
+
+def table_exists(table_path, storage_options):
+    # this is bad, i have to find the way to see if there is table behind path
+    try:
+        DeltaTable(table_path, storage_options=storage_options)
+    except Exception:
+        return False
+    return True
+
+
+## TODO
+# add partition writing
+# add optimization, vacumm options to automatically run before each run ?
```

Comment:
The plan is to make these things configurable on the delta table itself which then should handle at which interval to vacuum or optimize, similar to spark-delta.

Reply:
I am courious about that one is there some docu?

Reply:
Not yet :) Was just sharing what I was planning to add